src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs

```diff
@@ -0,0 +1,352 @@
+            testResult.ErrorMessage = e.Result.ErrorMessage;
+
+            var executionId = this.GetExecutionId(e.Result);
+            var parentExecutionId = this.GetParentExecutionId(e.Result);
+
+            this.TotalTests++;
+            testResult.resultOutcome = e.Result.Outcome;
+            if (e.Result.Outcome == TestOutcome.Failed)
+            {
+                this.FailedTests++;
+            }
+            else if (e.Result.Outcome == TestOutcome.Passed)
+            {
+                this.PassedTests++;
+            }
+            else if (e.Result.Outcome == TestOutcome.Skipped)
+            {
+                this.SkippedTests++;
+            }
+
+            if (parentExecutionId == Guid.Empty)
+            {
+                TestResults.Results.Add(testResult);
+            }
+
+            Results.TryAdd(executionId, testResult);
+            if (parentExecutionId != Guid.Empty)
+            {
+                this.AddToParentResult(parentExecutionId, testResult);
+            }
+        }
+
+        private void AddToParentResult( Guid parentExecutionId, TestResult testResult)
+        {
+            // Needs to be tested!
+            TestResult ParentTestResult;
+            this.Results.TryGetValue(parentExecutionId, out ParentTestResult);
+
+            if (ParentTestResult.innerTestResults == null)
+                ParentTestResult.innerTestResults = new List<TestResult>();
+
+            ParentTestResult.innerTestResults.Add(testResult);
+        }
+
+        /// <summary>
+        /// Creates a Summary of tests and Popultes the html file by transforming the xml file with help of xslt file
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        internal void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
+        {
+            TestResults.Summary = new TestRunSummary
+            {
+                FailedTests = this.FailedTests,
+                PassedTests = this.PassedTests,
+                TotalTests = this.TotalTests
+            };
+
+            if (this.parametersDictionary != null)
+            {
+                var isLogFileNameParameterExists = this.parametersDictionary.TryGetValue(HtmlLoggerConstants.LogFileNameKey, out string logFileNameValue);
+                if (isLogFileNameParameterExists && !string.IsNullOrWhiteSpace(logFileNameValue))
+                {
+                    this.htmlFilePath = Path.Combine(this.TestResultsDirPath, logFileNameValue);
+                }
+            }
+
+            this.PopulateHtmlFile();
+        }
+
+        private void PopulateHtmlFile()
+        {
+            fileName = String.Format(CultureInfo.CurrentCulture, "{0}_{1}_{2}", Environment.GetEnvironmentVariable("UserName"), Environment.MachineName, FormatDateTimeForRunName(DateTime.Now));
+            xmlFilePath = this.GetFilePath("xml",this.fileName);
+            Stream xmlStream = null;
+
+            try
```

Comment:
use Using block for filestreams [https://stackoverflow.com/questions/7524903/should-i-call-close-or-dispose-for-stream-objects](https://stackoverflow.com/questions/7524903/should-i-call-close-or-dispose-for-stream-objects)